### PR TITLE
Nb #67 cc index

### PR DESCRIPTION
**Pull request by nyxathid**
We now can:
- query CC index for any URL *u*
- get all paths to WET files that contain a page located at *u*
- get one of the files and extract all pages located at *u*

This is implemented in `getWebsiteByURL()` in `test-runs.ts`. We are already using the new `WetManager`, so the second run is much faster.

### CODE REVIEW

File: src/cc-index.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+    /**
+     * [HELPER FUNCTION]
+     * Query the CC index with a URL and call the callback with the RAW response string.
+     *
+     * @param lookupURL             URL to look up in the CC index
+     * @param callback              (optional) will be called with the response body string; if not provided -> output to console
+     * @param ccIndexPageURL        (optional) base URL of the CC index page; if not provided -> defaultCCIndex is used
+     */
+    public static lookUpURL(lookupURL : string,
```
Comment (lgfrbcsgo) [inline src/cc-index.ts]: Set the method to `private` if it's only a helper function? That way we won't get confused what the main method is when using this class.

File: src/cc-index.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    /**
+     * [HELPER FUNCTION]
+     * Convert each line of the response to a CCIndexResponse object and return and array of those objects.
+     *
+     * @param resp      CC index response string
+     * @returns {Array} array of response objects
+     */
+    public static parseStringToCCIndexResponse(resp : string) : Array<CCIndexResponse> {
```
Comment (lgfrbcsgo) [inline src/cc-index.ts]: Just like the other helper function.

File: src/cc-index.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+     * All paths are returned as a string array (no duplicates). Each returned path should start with "crawl-data/...".
+     *
+     * This function also checks if the lookup URL is included in the CC index response.
+     * If this is not the case, a warning will be shown.
+     *
+     * @param lookupURL             URL to look up
+     * @param resObjs               parsed response objects from the CC index
+     * @returns {Array<string>}     (relative) paths to WET files that contain websites with given url
+     */
+    public static constructWETPaths(lookupURL : string, resObjs : Array<CCIndexResponse>) : Array<string> {
```
Comment (lgfrbcsgo) [inline src/cc-index.ts]: Same as above 😎

**Comment by nyxathid** [comment]
Made helpers private
Used to test the separately from outside, but there is no need for that anymore

**Comment by kordianbruck** [comment]
issue #67

**Pull request merged**